src/figuresClasses.ts

```diff
@@ -1,19 +1,97 @@
-export interface Figure {
+enum Shape {
+  Triangle = 'triangle',
+  Circle = 'circle',
+  Rectangle = 'rectangle',
+}
 
+enum Color {
+  Red = 'red',
+  Green = 'green',
+  Blue = 'blue',
 }
 
-export class Triangle {
+export interface Figure {
+  shape: Shape;
+  color: Color;
 
+  getArea(): number;
 }
 
-export class Circle {
+const figureChecking = (...sides: number[]): boolean => {
+  return sides.some((side) => (side <= 0));
+};
+
+export class Triangle implements Figure {
+  public shape = Shape.Triangle;
+
+  constructor(
+    public color: Color,
+    public sideA: number,
+    public sideB: number,
+    public sideC: number,
+  ) {
+    if (figureChecking(sideA, sideB, sideC)) {
+      throw new Error('Passed value must be more than 0');
+    }
+
+    if (
+      sideA + sideB <= sideC
+      || sideB + sideC <= sideA
+      || sideC + sideA <= sideB
+    ) {
+      throw new Error('Side X is longer than sum of Y and Z');
+    }
+  }
 
+  getArea(): number {
+    const semiPerimeter = (this.sideA + this.sideB + this.sideC) / 2;
+
+    const area = Math.sqrt(
+      semiPerimeter
+        * (semiPerimeter - this.sideA)
+        * (semiPerimeter - this.sideB)
+        * (semiPerimeter - this.sideC),
+    );
+
+    return Math.floor(area * 100) / 100;
```

Comment:
you round down to hundredths on each getArea, maybe better create a separate method for it?